### PR TITLE
Allow users to save SDXL LoRA weights for only one text encoder

### PR DESCRIPTION
**Pull request by dulacp**
# What does this PR do?

Allow users to save only the LoRA weights for one text encoder if desired, often used in the Dreambooth realm to avoid overcooking the model too quickly.

Also, the method checks if at least one of `unet`, `text_encoder` or `text_encoder_2` lora weights is passed, which was not reflected in the implementation.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

I'm thinking about @sayakpaul since it's a topic close to training examples.

### CODE REVIEW

File: src/diffusers/loaders/lora.py [hunk 1]
```diff
@@ -1389,6 +1389,9 @@ def save_lora_weights(
             text_encoder_lora_layers (`Dict[str, torch.nn.Module]` or `Dict[str, torch.Tensor]`):
                 State dict of the LoRA layers corresponding to the `text_encoder`. Must explicitly pass the text
                 encoder LoRA state dict because it comes from 🤗 Transformers.
+            text_encoder_2_lora_layers (`Dict[str, torch.nn.Module]` or `Dict[str, torch.Tensor]`):
+                State dict of the LoRA layers corresponding to the `text_encoder_2`. Must explicitly pass the text
+                encoder LoRA state dict because it comes from 🤗 Transformers.
```
[the remark this answers is not in the document]
Comment (dulacp) [reply src/diffusers/loaders/lora.py:1394]: Yes, this is what I aimed to fix.

Just added the missing parameter in the documentation (to match the signature) and split the following if condition: https://github.com/huggingface/diffusers/blob/7e39516627c69b71f8b21a2b53689028d4733b72/src/diffusers/loaders/lora.py#L1418

[the remark this answers is not in the document]
Comment (sayakpaul) [reply src/diffusers/loaders/lora.py:1394]: Thanks for explaining! LGTM.

[the remark this answers is not in the document]
Comment (dulacp) [reply src/diffusers/loaders/lora.py:1394]: Thanks for looking into it in record time :)
Cheers

**Comment by yiyixuxu** [comment]
can you fix the tests?

**Comment by yiyixuxu** [comment]
ohh the failing test actually might not be relevant here - will take a look on our end!

**Comment by dulacp** [comment]
Yes, I realized that it seems to be related to something else. Can you re-run the failing job?

**Review by yiyixuxu — APPROVED** [review]

**Comment by yiyixuxu** [comment]
thank you!

**Pull request merged**